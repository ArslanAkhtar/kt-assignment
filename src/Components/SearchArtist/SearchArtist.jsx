import React, { Fragment, useState } from "react";
import { Row, Col } from "antd";
import SearchField from "../Subcomponents/SearchField";
import ArtistCard from "../Subcomponents/ArtistCard";
import EventCards from "../Subcomponents/EventCards";

const SearchArtist = () => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(false);
  const [artistName, setArtistName] = useState(null);
  const [eventsData, setEventsData] = useState(null);
  return (
    <Fragment>
      <Row>
        <Col span={24}>
          <SearchField
            setData={setData}
            setLoading={setLoading}
            artistName={artistName}
            setArtistName={setArtistName}
            setEventsData={setEventsData}
          />
        </Col>
      </Row>
      {data && (
        <ArtistCard
          data={data}
          loading={loading}
          artistName={artistName}
          setEventsData={setEventsData}
        />
      )}
      {eventsData && <EventCards eventsData={eventsData} />}
    </Fragment>
  );
};

export default SearchArtist;
