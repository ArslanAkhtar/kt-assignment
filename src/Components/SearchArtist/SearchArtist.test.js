import { shallow } from "enzyme";
import React from "react";
import SearchArtist from "./SearchArtist";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  shallow(<SearchArtist />);
});

it("renders correctly with Snapshot", () => {
  const tree = shallow(<SearchArtist />);
  expect(toJson(tree)).toMatchSnapshot();
});

describe("Contain Name", () => {
  it("contains Name", () => {
    const wrapper = shallow(<SearchArtist />);
    expect(wrapper.contains("Search Artist"));
  });
});
