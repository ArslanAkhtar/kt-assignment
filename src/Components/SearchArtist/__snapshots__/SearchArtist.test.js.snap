// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Snapshot 1`] = `
<Fragment>
  <Row>
    <Col
      span={24}
    >
      <Memo(SearchField)
        artistName={null}
        setArtistName={[Function]}
        setData={[Function]}
        setEventsData={[Function]}
        setLoading={[Function]}
      />
    </Col>
  </Row>
</Fragment>
`;
